internal/metrics/metrics_test.go

```
@@ -0,0 +1,52 @@
+)
+
+func TestNewAndHandler(t *testing.T) {
+	m, err := New()
+	require.NoError(t, err)
+	require.NotNil(t, m)
+
+	m.ScanCounter.Add(context.Background(), 1, metric.WithAttributes())
+	m.ScanDuration.Record(context.Background(), 0.5, metric.WithAttributes())
+	m.RejectCounter.Add(context.Background(), 1, metric.WithAttributes())
+
+	req := httptest.NewRequest("GET", "/metrics", nil)
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)